app/controllers/users_controller.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def show
+    @user = User.find_by(id: params[:id])
+    @upcoming_events = @user.attended_events.upcoming
+    @past_events = @user.attended_events.past
+  end
+
+  def index
+    @users = User.all
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      log_in @user
+      redirect_to users_path
+    else
+      redirect_to new_user_path
+    end
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit(:name, :email, :password)
+  end
```

Comment:
- [ ] Signing up gives `unkown attribute 'password' for User.`. That's because user model doesn't have attribute with the name `password` and you don't have bcrypt gem working `has_secure_password`. I suggest that you read through [secure password](https://api.rubyonrails.org/classes/ActiveModel/SecurePassword/ClassMethods.html) to make it work.
HINT: You haven't included `has_secure_password` in your user model.

Reply:
Fixed Signing up user error `unkown attribute 'password' for User.`
add `has_secure_password` to  `User` model